drivers/gpio/gpio_davinci.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_davinci_gpio
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/arch/common/sys_bitops.h>
+#include <zephyr/devicetree.h>
+
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+/* Helper Macros for GPIO */
+#define DEV_GPIO_CFG(dev)	\
+		((const struct gpio_davinci_config * const)((dev)->config))
+#define DEV_GPIO_CFG_BASE(dev) \
+		((struct gpio_davinci_regs *)(DEV_GPIO_CFG(dev)->port_base))
+
+#define GPIO_DAVINCI_DIR_RESET_VAL	(0xFFFFFFFF)
+
+struct gpio_davinci_regs {
+	uint32_t dir;
+	uint32_t out_data;
+	uint32_t set_data;
+	uint32_t clr_data;
+	uint32_t in_data;
+	uint32_t set_ris_trig;
+	uint32_t clr_ris_tirg;
+	uint32_t set_fal_trig;
+	uint32_t clr_fal_trig;
+	uint32_t intstat;
+};
+
+struct gpio_davinci_data {
+	struct gpio_driver_data common;
+	sys_slist_t cb;
+};
+
+struct gpio_davinci_config {
+	void (*bank_config)(const struct device *dev);
+	struct gpio_driver_config common;
+	uint32_t port_base;
+	uint32_t port_num;
+};
+
+static int gpio_davinci_configure(const struct device *dev, gpio_pin_t pin,
+									gpio_flags_t flags)
+{
+	volatile struct gpio_davinci_regs *regs = DEV_GPIO_CFG_BASE(dev);
+
+	if ((flags & GPIO_SINGLE_ENDED) != 0) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & (GPIO_PULL_UP | GPIO_PULL_DOWN)) != 0) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_OUTPUT) != 0) {
+		if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0) {
+			regs->set_data = BIT(pin);
+		} else {
+			regs->clr_data = BIT(pin);
+		}
+
+		regs->dir &= (~(BIT(pin)));
+
+	} else {
+		regs->dir |= (BIT(pin));
+	}
+
+	return 0;
+}
+
+static int gpio_davinci_port_get_raw(const struct device *dev,
+									gpio_port_value_t *value)
```

Comment:
Looks like the way you broke the arguments made it actually worser and will probably exceed 80 characters. These arguments should be placed close to the opening bracket / first argument.
```suggestion
static int gpio_davinci_port_get_raw(const struct device *dev,
					gpio_port_value_t *value)
```

Reply:
@dnltz I will align the arguments to the opening bracket (my vim editor has tabspace set to 2 spaces, so the alignment looked fine in my editor, I will change it).

Reply:
FWIW tab space is 8 for Zephyr, and max line length is 100 characters (not 80).